### PR TITLE
Use properties instead of ivars in macOS client

### PR DESCRIPTION
**Pull request by DevilDimon**
In 'Adopting Modern Objective-C' Apple suggests using properties instead of ivars wherever possible. It brings many benefits and also modernises the code – you can hardly see any ivars in modern objc codebases these days.

This PR:

- Replaces all ivars with properties of the same semantics
- Moves `IBOutlet` properties to implementation files to improve encapsulation
- Replaces `Private` categories with extensions where needed (categories don't support stored properties)

### CODE REVIEW

**Review by ckerr — DISMISSED** [review]
Thanks for the PR!

1. FTBFS in CI:

```txt
  /Users/teamcity/buildAgent1/work/1f3bff43dd55cfe9/macosx/Controller.mm:5340:5: error: use of undeclared identifier 'fQuitRequested'; did you mean '_fQuitRequested'?
19:09:22       fQuitRequested = YES;
```

2. Looks like clang-tidy is unhappy about the double linefeed between `- (void)openURLCancelEndSheet:(id)sender;` and `@end` at the bottom of `macosx/URLSheetWindowController.h`. If you remove one of those lines, I _think_ the Code Style bot will go green.

**Comment by livings124** [comment]
Thanks for doing this! Because of the large change, we'll merge this in after the next release.

**Comment by GScottElshaw** [comment]
Can someone explain why this modernisation can't be incorporated until later, it just seems like progress to me? Is there anything here likely to irreparably break anything @DevilDimon can't help fix? It feels like a gift to have a contributor who especially wants to help on the Mac side of things right now while Charles is doing the hard yards with all of the C++ modernising.

**Comment by livings124** [comment]
Sorry, the wait was based on an earlier estimate of our release schedule. With the current plan this will have time to bake, so we can go ahead and merge (once the conflict is resolved)

**Review by livings124 — APPROVED** [review]

File: macosx/PrefsController.mm [hunk 1]
```diff
@@ -109,19 +168,19 @@ - (instancetype)initWithHandle:(tr_session*)handle
...
 
         //reset old Sparkle settings from previous versions
-        [fDefaults removeObjectForKey:@"SUScheduledCheckInterval"];
-        if ([fDefaults objectForKey:@"CheckForUpdates"])
+        [_fDefaults removeObjectForKey:@"SUScheduledCheckInterval"];
+        if ([_fDefaults objectForKey:@"CheckForUpdates"])
         {
-            [[SUUpdater sharedUpdater] setAutomaticallyChecksForUpdates:[fDefaults boolForKey:@"CheckForUpdates"]];
-            [fDefaults removeObjectForKey:@"CheckForUpdates"];
+            //[[SUUpdater sharedUpdater] setAutomaticallyChecksForUpdates:[fDefaults boolForKey:@"CheckForUpdates"]];
```
Comment (Coeur) [inline macosx/PrefsController.mm:182]: @DevilDimon what happened here? Why was this line commented out?

**Pull request merged**